### PR TITLE
Azure queue streams configuration

### PR DESCRIPTION
**Pull request by rallets**
Following discussion #6300 
Add a new page with some details about usage, configuration and implementation of Azure Queue Streams.

### CODE REVIEW

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+This section provides some details about the usage, configuration and implementation of **Orleans Azure Queue Streams**.
+More details are available in the streaming tests, which contain most of the configuration options, specifically [**`AQClientStreamTests`**](https://github.com/dotnet/orleans/tree/master/test/Extensions/TesterAzureUtils/Streaming/AQClientStreamTests.cs), [**`AQSubscriptionMultiplicityTests`**](https://github.com/dotnet/orleans/tree/master/test/Extensions/TesterAzureUtils/Streaming/AQSubscriptionMultiplicityTests.cs), and the extension functions for [**`IAzureQueueStreamConfigurator`**](https://github.com/dotnet/orleans/tree/master/src/Azure/Orleans.Streaming.AzureStorage/Providers/Streams/AzureQueue/AzureQueueStreamBuilder.cs)  and [**`ISiloPersistentStreamConfigurator`**](https://github.com/dotnet/orleans/tree/master/src/Orleans.Runtime.Abstractions/Streams/ISiloPersistentStreamConfigurator.cs).
+
+## Overview
+
+Orleans Azure Queue requires the package **Microsoft.Orleans.Streaming.AzureStorage** and the silo configuration at startup allow to specify the connection string
```
Comment (jason-bragg) [inline src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: > "and the silo configuration at startup allow to specify the connection string" - I think I understand what's being stated here, but unclear.  Can you take another pass at the wording.  Thanks

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+``` csharp
+hostBuilder
+  .AddAzureQueueStreams<AzureQueueDataAdapterV2>("AzureQueueProvider", optionsBuilder => optionsBuilder.Configure(options => {
+    options.ConnectionString = "xxx";
+  }))
+  // also a PubSubStore is needed, as example using Azure Table Storage
```
Comment (jason-bragg) [inline src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: Pubsub needs only be configured using default or explicit subscriptions.  It's not needed if using only implicit subscriptions only.  Not sure how much we want to get into subscription configuration in this section, but maybe add a 'by default' qualifier to the the comment?

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  .AddAzureTableGrainStorage("PubSubStore", options => {
+    options.ConnectionString = "xxx";
+  })
+```
+
+The pulling agents will pull repeatedly until there are no more messages on a queue, then **delay** for a configurable period before continuing to pull. This happen for **each queue**.
```
Comment (jason-bragg) [inline src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: > "This happen for **each queue**."

This happen**s** fo...?

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+layout: page
+title: Azure Queue Streams Implementation Details
+---
+
+# Orleans Azure Queue Streams Implementation Details
+
+Each stream provider (Azure Queues, EventHub, SMS, SQS, ...) has it's own queue specific details and configuration.
+This section provides some details about the usage, configuration and implementation of **Orleans Azure Queue Streams**.
+More details are available in the streaming tests, which contain most of the configuration options, specifically [**`AQClientStreamTests`**](https://github.com/dotnet/orleans/tree/master/test/Extensions/TesterAzureUtils/Streaming/AQClientStreamTests.cs), [**`AQSubscriptionMultiplicityTests`**](https://github.com/dotnet/orleans/tree/master/test/Extensions/TesterAzureUtils/Streaming/AQSubscriptionMultiplicityTests.cs), and the extension functions for [**`IAzureQueueStreamConfigurator`**](https://github.com/dotnet/orleans/tree/master/src/Azure/Orleans.Streaming.AzureStorage/Providers/Streams/AzureQueue/AzureQueueStreamBuilder.cs)  and [**`ISiloPersistentStreamConfigurator`**](https://github.com/dotnet/orleans/tree/master/src/Orleans.Runtime.Abstractions/Streams/ISiloPersistentStreamConfigurator.cs).
```
Comment (jason-bragg) [inline src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: > "More details are available in the streaming tests, which contain most of the configuration options, specifically"

- Let's make sure readers understand that this section is not comprehensive.  Maybe something like:

"This section is not comprehensive, and more details are available in the streaming tests, which ..."

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md (hunk 1 above)
Comment (rallets) [reply src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: is it more clear now?

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md (hunk 2 above)
Comment (rallets) [reply src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: Is the comment better now?
Interesting, I believed it was needed by design to store the subscriptions. I looked to the documentation by I haven't found mention that this is "optional". I don't think this page is the right place to dig in this concept, as it is a more global.

Comment (rallets) [reply src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: If this is something that you want to be covered in an other page, I can try to submit another PR.

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md (hunk 3 above)
Comment (rallets) [reply src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: Is this sentence better now?

File: src/Documentation/implementation/streams_implementation/azure_queue_streams.md (hunk 4 above)
Comment (rallets) [reply src/Documentation/implementation/streams_implementation/azure_queue_streams.md]: try now

**Comment by rallets** [comment]
@jason-bragg let me know if I need to do more, any feedback is welcome.

**Comment by jason-bragg** [comment]
Thanks @rallets
Great work!

**Comment by rallets** [comment]
Thanks @jason-bragg for your support, really appreciated

**Pull request merged**